### DISCUSSION TITLE
RuntimeError: Only Tensors of floating point and complex dtype can require gradients

### DISCUSSION THREAD

**Discussion by fangqiangdev** [chatchat-space/Langchain-Chatchat#744] [Q&A]
我改了moel_config.py中2个地方的配置。然后启动报错如下，有遇到相同问题吗，求解答。

系统：macos （使用cpu）
python: conda + python3.8.16

![image](https://github.com/imClumsyPanda/langchain-ChatGLM/assets/105053537/2ddc54a5-4639-4986-8703-e5cd71432c12)

![image](https://github.com/imClumsyPanda/langchain-ChatGLM/assets/105053537/693159e0-fbc1-485a-a993-03d946ec12a9)

**Comment by xiexj2** [comment]
This issue can also occur when using chatglm-6b-int4 or chatglm2-6b-int4, but not when using chatglm2-6b.

**Reply by molihan** [reply]
In that case, implementing the init_model method instead.
